src/server/scripts/EasternKingdoms/Deadmines/instance_deadmines.cpp

```diff
@@ -74,10 +77,16 @@ class instance_deadmines : public InstanceMapScript
             ObjectGuid DefiasCompanionGUID;
 
             uint32 State;
+            uint32 MrSmiteGUID;
             uint32 CannonBlast_Timer;
             uint32 PiratesDelay_Timer;
             ObjectGuid uiSmiteChestGUID;
 
+            void OnCreatureCreate(Creature* creature) override
+            {
+                MrSmiteGUID = creature->GetGUID();
```

Comment:
switch (creature->GetEntry()) case urNpcEntry: ...

Right now your hook looks like this: Assign guid from any creature created in instance to MrSmiteGUID